### PR TITLE
feat: add postcss support to default template

### PR DESCRIPTION
**Pull request by rishabh3112**
**What kind of change does this PR introduce?**
Add postcss support as well to default template

**Did you add tests for your changes?**
WIP

**If relevant, did you update the documentation?**
N/A

**Summary**
Add PostCSS support as well to default template because it is popular.

**Does this PR introduce a breaking change?**
No

**Other information**
/cc @alexander-akait

### CODE REVIEW

**Review by evenstensberg — COMMENTED** [review]
Postcss is an addition to sass, css etc and should be added automatically

**Comment by rishabh3112** [comment]
> Postcss is an addition to sass, css etc and should be added automatically

Thanks for review. Here are my opinions:
- Postcss with sass is certainly not a popular option, people either use sass or move all together to using Postcss with css by adding relevent postcss plugins.

- Adding it automatically is not correct as well, many people want to use just css as it works better for many with less dependancy and less build time. (This includes me when I do small uni projects)

Nevertheless, I think I should tag @alexander-akait here as he knows more about how most projects use webpack for css preprocessing being maintainer of many loaders for them.

File: packages/generators/init-template/default/postcss.config.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    // Add you postcss configuration here
+    // Learn more about it at https://github.com/webpack-contrib/postcss-loader#config-files
+    plugins: [['postcss-preset-env']],
```
Comment (alexander-akait) [inline packages/generators/init-template/default/postcss.config.js:4]: I think better put `autoprefixer` here, it is more popular

File: packages/generators/src/handlers/default.ts [hunk 2]
```diff
@@ -56,7 +56,7 @@ export async function questions(self: CustomGenerator, Question: Record<string,
         self,
         'cssType',
         'Which of the following CSS solutions do you want to use?',
-        ['none', 'CSS', 'SASS', 'LESS', 'Stylus'],
+        ['none', 'CSS', 'SASS', 'LESS', 'Stylus', 'PostCSS'],
```
Comment (alexander-akait) [inline packages/generators/src/handlers/default.ts:59]: I think we need refactor here, ideally:
- Do you want to use CSS? - None/Yes (style-loader and css-loader), will be great to add `mini-css-extract-plugin` when mode is production, but we can do it in separate PR
- Do you want to use `CSS` preprocessor: `SASS`/`Less`/`Stylus`
- Do you want to use `CSS` postprocessolor : `postcss`

What do you think?

Comment (rishabh3112) [reply packages/generators/src/handlers/default.ts:59]: `mini-css-extract-plugin` seems like like a very good idea, will add this in.
Meanwhile, I couldn't understand how adding three separate questions for it seems to bring any value as it is always nice to keep things simple and quick to use. A little more explanation on your intentions would be nice.

Comment (alexander-akait) [reply packages/generators/src/handlers/default.ts:59]: Some developers use `postcss`, some not, maybe we can do it in our questions. 

Do you need CSS?
- None
- CSS
- CSS and PostCSS
- CSS and SASS (for less and Stylus too)
- CSS, PostCSS and SASS (for less and Stylus too)

I was thinking about separate this into different questions... There are a lot of answers...

Comment (rishabh3112) [reply packages/generators/src/handlers/default.ts:59]: Hmm, I understood. I think I would do that in seperate PR. Currently in this PR, I am adding PostCSS  with CSS only.

Comment (alexander-akait) [reply packages/generators/src/handlers/default.ts:59]: Yep, let's do it in separate PR

File: packages/generators/init-template/default/postcss.config.js (hunk 1 above)
Comment (rishabh3112) [reply packages/generators/init-template/default/postcss.config.js:4]: I found one more thing while exploring, `autoprefixer` is already included in `postcss-preset-env` (https://github.com/csstools/postcss-preset-env#autoprefixer).

Comment (alexander-akait) [reply packages/generators/init-template/default/postcss.config.js:4]: `postcss-preset-env` is very poorly updated and contains rather controversial solutions, if someone wants to add it, he can do it very easily, most developers do not use it due a lot of bugs and problems

**Review by alexander-akait — DISMISSED** [review]
cc @webpack/cli-team

**Review by snitin315 — APPROVED** [review]

**Review by alexander-akait — APPROVED** [review]
